include/stl2/view/const.hpp

```diff
@@ -0,0 +1,276 @@
+// cmcstl2 - A concept-enabled C++ standard library
+//
+//  Copyright Dvir Yitzchaki 2019
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/caseycarter/cmcstl2
+//
+#ifndef STL2_VIEW_CONST_HPP
+#define STL2_VIEW_CONST_HPP
+
+#include <stl2/detail/fwd.hpp>
+#include <stl2/detail/range/concepts.hpp>
+#include <stl2/view/view_interface.hpp>
+#include <stl2/view/all.hpp>
+
+STL2_OPEN_NAMESPACE
+{
+
+	namespace ext
+	{
+	template <input_range R>
+	requires view<R> 
+	class const_view : public view_interface<const_view<R>>
+	{
+		template <bool>
+		struct __iterator;
+
+	public:
+		const_view() = default;
+		constexpr explicit const_view(R base) : base_(std::move(base)) {}
+
+		constexpr auto base() const noexcept -> R { return base_; }
+
+		constexpr __iterator<false> begin()
```

Comment:
You'll need to adda requirement for  `detail::simple_view` on each of the non-const overloads.

Reply:
Sorry, I meant `!simple_view`.